### PR TITLE
feat(ts): use jsdelivr to check for d.ts

### LINKED ISSUE #511: Detect packages with TypeScript but without a `types` field in package.json
Would be nice if the search would correctly detect a package as TypeScript in the case where a package has a `index.d.ts` file in root, but not a `types` field in `package.json`. That's a common occurrence and it's allowed by TypeScript to omit the `types` field if the definition file uses the name `index.d.ts`.

Context: https://github.com/algolia/npm-search/pull/407#issuecomment-584338683

### PR DESCRIPTION
**Pull request by bodinsamuel**
A bit copied from #407 

Should fix #511

Stop using unpkg and reuse JsDelivr filelist to get changelog and ts def

### CODE REVIEW

**Review by Haroenv — APPROVED** [review]

File: src/typescript/index.ts [hunk 1]
```diff
@@ -73,22 +74,17 @@ export async function getTypeScriptSupport(
...
-  // TO DO: replace this with a list of files check
-  if (pkg.types.ts !== 'definitely-typed' && pkg.types.ts.possible === true) {
-    const resolved = await fileExistsInUnpkg(
-      pkg.name,
-      pkg.version,
-      pkg.types.ts.dtsMain
-    );
-    if (resolved) {
-      return { types: { ts: 'included' } };
+  for (const file of filelist) {
```
[the remark this answers is not in the document]
Comment (bodinsamuel) [reply src/typescript/index.ts:77]: it's the habits here, no strong preference either

File: src/changelog.ts [hunk 2]
```diff
@@ -141,19 +143,24 @@ export async function getChangelog(
...
   Array<{
     changelogFilename: string | null;
   }>
 > {
   const start = Date.now();
 
-  const all = await Promise.all(pkgs.map(getChangelog));
+  const all = await Promise.all(
+    pkgs.map((pkg, index) => {
+      return getChangelog(pkg, filelists[index] || []);
```
[the remark this answers is not in the document]
Comment (bodinsamuel) [reply src/changelog.ts:161]: its not needed but clearer on the expectation, it's also the habits does not influence end results

File: src/saveDocs.ts [hunk 3]
```diff
@@ -57,12 +57,16 @@ export default async function saveDocs({
...
-    getTSSupport(pkgs),
+    jsDelivr.getAllFilesList(pkgs),
+  ]);
+
+  const [changelogs, ts] = await Promise.all([
+    getChangelogs(pkgs, filelists),
+    getTSSupport(pkgs, filelists),
   ]);
 
   const start = Date.now();
```
[the remark this answers is not in the document]
Comment (bodinsamuel) [reply src/saveDocs.ts:72]: it was like that before, I guess this metric is not really relevant as it only calc the spreading time.
Every other calls are tracked independently

File: src/typescript/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bodinsamuel) [reply src/typescript/index.ts:77]: the expectation is that I can not return so it would had an additional if, so kiff - kiff

**Comment by bodinsamuel** [comment]
Saved a couple of seconds from TS check 🙆🏻‍♂️ 
Screenshot 2021-07-05 at 19 09 50

**Pull request merged**